src/mds/Beacon.cc

```diff
@@ -173,7 +173,11 @@ void Beacon::send_and_wait(const double duration)
   while (!seq_stamp.empty() && seq_stamp.begin()->first <= awaiting_seq) {
     auto now = clock::now();
     auto s = duration*.95-std::chrono::duration<double>(now-start).count();
-    if (s < 0) break;
+    if (s < 0) {
+      //missed beacon ACKs
+      missed_beacon_ack_dump = true;
```

Comment:
Do you think collection additional stats like the sequence number of the beacon to get acked would be useful in the dump?

Reply:
Also, should something similar be done in the monitor. Note that this is mostly as an aid for debugging, and when debugging missed beacons, we would generally consult the monitor logs to cross-verify and infer reasons for missing beacons acks.

Reply:
> Do you think collection additional stats like the sequence number of the beacon to get acked would be useful in the dump?

Currently it dumps what's given by `g_ceph_context->_log->dump_recent()` call. Will check on the additional stats and dump them separately. Won't that be already present in the current dump text given by `g_ceph_context->_log->dump_recent()` ?


> Also, should something similar be done in the monitor. Note that this is mostly as an aid for debugging, and when debugging missed beacons, we would generally consult the monitor logs to cross-verify and infer reasons for missing beacons acks.

That's right. All these extra-ordinary events has corresponding logs in the monitor also. But they seem to be very less (short description) compared to the corresponding in-memory dump and that's something already included in the in-memory log dump itself.